server/hl7/hl7.controller.js

```diff
@@ -108,5 +109,63 @@ function getUserFiles(req, res, next) {
   return next(err);
 }
 
+function getMessageByIndexOrId(req, res, next) {
+  let err;
+  const fileId = req.params.fileId;
+  if (!mongoose.Types.ObjectId.isValid(req.params.indexWithinFileOrId)) { // treat as index in file
+    const messageIndex = req.params.indexWithinFileOrId;
+    if (!Number.isInteger(parseInt(messageIndex, 10))) {
+      err = new APIError(`Invalid Message Index: ${messageIndex}`, httpStatus.BAD_REQUEST);
+      return next(err);
+    }
+    return _getMessageByIndex(messageIndex, fileId)
+      .then((message) => {
+        if (message.length > 0) res.status(httpStatus.OK).json(message);
+        else res.status(httpStatus.NOT_FOUND).json('Message Not found. Check index number & File ID');
```

Comment:
I prefer curly braces around all if/elses
also indentation isn't what I wouldn't done but also not bad - maybe we should bring in `prettier` or something (out of scope for this PR, just a thought)

Reply:
As you desire your highness lol